Implemented the Auth module, login and logout components and cleaned up app.component

ravisaxena23 opened the pull request:
**What type of PR is this?**
 /kind new feature
 /kind bug fix
 /kind cleanup
 /kind enhancement

**What this PR does / why we need it**:
We have optimized the login process. added the login code to the new auth module, and cleaned the app component.
This PR also fixes the issue present on #153. this is because we have moved the login code to a different module so now the router does not navigate to home every time.
This PR  Implements Auth Module, login component, and logout component.

**Which issue(s) this PR fixes**:
Fixes #650 
Fixes #153

**Test Report Added?**:
TESTED
I have tested the login and logout and some scenarios as well.  I am sorry but I am unable to test consts file changes currently please help me with it.

**Test Report**
![Screenshot (7)](https://user-images.githubusercontent.com/44205807/165153298-dc43071f-0f33-47f2-bb2d-be3f08f7febb.png)
![Screenshot (8)](https://user-images.githubusercontent.com/44205807/165153311-02629c48-5fae-47ea-bf25-ee2b5daac1d6.png)
![Screenshot (9)](https://user-images.githubusercontent.com/44205807/165153316-44b16c8c-7cac-493b-a3ae-af073285f03e.png)
![Screenshot (10)](https://user-images.githubusercontent.com/44205807/165153321-f10e2b74-1f3d-4844-8726-1a184138727c.png)


**Special notes for your reviewer**:
I have tested the login and logout and some scenarios as well.  I am sorry but I am unable to test consts file changes currently please help me with it. 
I am sorry my linter has made small changes.

anvithks reviewed (changes requested):
Some initial observations. Could you PTAL

ravisaxena23 commented on src/app/business/auth/login/login.component.ts:
```diff
@@ -0,0 +1,356 @@
+                this.paramStor.AUTH_TOKEN(r.headers.get('x-subject-token'));
+                this.paramStor.CURRENT_TENANT(project.name + "|" + project.id);
+                this.paramStor.CURRENT_USER(user.name + "|" + user.id);
+                this.paramStor.CURRENT_ROLE(JSON.stringify(roles))
```
remove this line this.paramStor.CURRENT_ROLE(JSON.stringify(roles))

ravisaxena23 commented on src/app/business/auth/logout/logout.component.html:
```diff
@@ -0,0 +1,3 @@
+<h1>
+  logout works!
```
removed dummy files from both login and logout

ravisaxena23 commented:
@anvithks  I have added a blank logout template as the template is required for the component. it was showing an error on the console If the file is not present. further, I have moved menu_items to the login component, if it is required we can create a separate issue for it.

anvithks commented:
Some more testing is needed to verify if any other pages are affected. 
But overall it looks like the Auth module has been implemented properly. 
Next we add the Auth guard to the routes.

ravisaxena23 commented:
Fixes #153

joseph-v reviewed (approved):
LGTM

anvithks reviewed (approved):
LGTM
